task_10/src/Favourites/Favourites.js

```diff
@@ -0,0 +1,78 @@
+require('./Favourites.scss');
+
+import { Component } from '../Framework';
+
+export default class Favourites extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			list: this.getFromLocalStorage('favourites')
+		}
+		this.host = document.createElement('div');
+		this.host.classList.add('favourites__container');
+		this.ul = document.createElement('ul');
+		this.ul.classList.add('favourites');
+		this.host.appendChild(this.ul);
+
+		this.handleClick = this.handleClick.bind(this);
+		this.ul.addEventListener('click', this.handleClick);
+
+		this.addButton = document.createElement('button');
+		this.addButton.classList.add('favourites__add-button', 'button');
+		this.addButton.addEventListener('click', () =>
+			this.add(this.props.city, 'favourites')
+		);
+		this.addButton.innerHTML = '<i>+</i>';
+		this.addButton.title = 'add to favourites';
+		this.host.appendChild(this.addButton);
+
+		this.clearButton = document.createElement('button');
+		this.clearButton.classList.add('favourites__clear-button', 'button');
+		this.clearButton.addEventListener('click', () =>
+			this.clear('favourites'));
+		this.clearButton.title = 'clear favourites'
+		this.clearButton.innerHTML = '<i>+</i>';
+		this.host.appendChild(this.clearButton);
+	}
+
+	render() {
+		this.ul.innerHTML = '';
+		const list = this.state.list;
+		for (var i = list.length - 1; i >= 0; i--) {
+			const city = list[i];
+			const li = `
+				<li class="favourites__city">
+					<a href="#">${city}</a>
+				</li>
+			`;
+			this.ul.insertAdjacentHTML('beforeend', li);
+		}
+		return [this.ul, this.addButton, this.clearButton];
+	}
+
+	add(item, key) {
+		const list = new Set(this.state.list);
+		if (list.has(item)) list.delete(item);
+		list.add(item);
+		localStorage[key] = JSON.stringify([...list]);
+		this.updateState({ list: [...list] });
```

Comment:
I don't understand why you doing such things here :)

If you need only uniq elements in the array, you can use `Array#includes` or `Array#indexOf`.

Reply:
First, I had an `Array`. But then you asked me if I could use a `Set` here.   
So I tried. Maybe I misunderstood you about the reason to use `Set`  😄

Reply:
My bad, I could forgot :)

Now you using set and array in the same time. Can you use only set?

Reply:
Ok, I checked - you can't "just" get `Set` from localStorage.  If you just use `JSON.parse` on the results of `getItem`, you will receive an array. 

Therefore, taking in to action necessity of working with localStorage, I can suggest two options:

1. You should return to array (sorry 👼 )
2. You still can use `Set`, but IMO, you should have some abstract util, which will convert sets to strings and vice-versa.  

For now, I think that array will be the best option.

P.S. But I still don't understand why you remove item, and then add new item? Why not just return from function, if such value already exists?

Reply:
It's ok. Some practice with a `Set` was a great option for me 😄 
I remove existing item because I want it to be in the end of the list.